src/coreclr/minipal/Unix/doublemapping.cpp

```diff
@@ -47,6 +47,90 @@ static const off_t MaxDoubleMappedSize = UINT_MAX;
 
 #endif // TARGET_OSX
 
+#if defined(TARGET_AMD64) && !defined(TARGET_OSX)
```

Comment:
Is there any technical issue that require us to restrict this run-time introspection? If we don't restrict it, then it will cover qemu cases for other archs as well (e.g. https://github.com/dotnet/runtime/issues/97729#issuecomment-2050507728), and prevent user from manually setting `DOTNET_EnableWriteXorExecute=1` to run .NET applications.

Reply:
I remember once @jkotas mentioned that we wanted to support rosetta2 on macOS as a first-class platform, and therefore we are handling it explicitly with `sysctl.proc_translated` in a few places (coreclr, nativelibs and shell scripts and even in official build infra for downloading x64 on arm64 in translated process). I think it might be better to bring this to `IsProcessTranslated` plan as well without the run-time detection? https://github.com/dotnet/runtime/blob/362a95d53d7266ee9b7452e351d07845dcc1edd2/src/coreclr/minipal/Unix/doublemapping.cpp#L28

Reply:
`sysctl.proc_translated` is not going to help. This is about running Linux docker containers. We do not know about `sysctl.proc_translated` equivalent for Linux docker containers running under Rosetta.

Reply:
Ah, I see, it makes sense. Could we enable this feature detection for other platforms as well (which will cover qemu scenarios)? (assuming the overhead due to this one-time check is negligible)

Reply:
@am11 I've tried to look at the discussion you have linked, but I have not seen any evidence of the culprit being inability to double map. But I might have missed that. Can you please point me to such information? The point is, if it is not a problem of double mapping, this check may not be able to check the other issue. 
When I started to investigate this issue, I actually still had a docker desktop that used QEMU for x64 emulation on arm64 and even with W^X disabled, attempt to build .NET runtime was failing at some point due to SIGSEGV. Only after that I've realized I need to upgrade docker to get the version that used Rosetta to be able to make it work even with W^X disabled.

Reply:
@janvorli, I was running into `qemu: uncaught target signal 11 (Segmentation fault)` which was fixed with `DOTNET_EnableWriteXorExecute=0`. Later dotnet publish  was failing a runtime assertion on qemu arm target. But to get to that late assertion `DOTNET_EnableWriteXorExecute=0` solution to sigsegv was the roadblock. In the context of this PR, I thought it might be useful to keep the condition a bit relaxed.

Reply:
@janvorli, I was running into `qemu: uncaught target signal 11 (Segmentation fault)` which was fixed with `DOTNET_EnableWriteXorExecute=0`. Later dotnet publish  was failing a runtime assertion on qemu arm target. But to get to that late assertion sigsegv was the roadblock that `DOTNET_EnableWriteXorExecute=0` fixed. In the context of this PR, I thought it might be useful to keep the condition a bit relaxed.